### ISSUE TITLE
Add feature of ComposePathEffect and CornerPathEffect

### ISSUE THREAD

**Issue by premingiet** [flutter/flutter#68081]
## Use case

There is no way or an equivalent solution for path smoothing in flutter. Like in android we have `ComposePathEffect, CornerPathEffect, DashPathEffect, DiscretePathEffect, PathDashPathEffect, SumPathEffect`  but i don't see any equivalent solution for flutter. How do i smooth the path that is drawn ?

## Proposal

Please look at this page and this is what i am imagining to do with flutter too
http://android-coding.blogspot.com/2014/05/composepatheffect-example.html

Google SKIA already has these. You can see from here
https://github.com/google/skia/tree/master/src/effects

and because all flutter renderings are handled by skia backend, so i feel this is not that tough to implement.

**Comment by darshankawar** [comment]
May be related to [this](https://github.com/flutter/flutter/issues/4858) closed issue.

**Comment by premingiet** [comment]
No, actually that issue [#4858](https://github.com/flutter/flutter/issues/4858) just discusses about dashed line. What i actually want is a path smoothing technique like this [path corner smoothing](http://android-coding.blogspot.com/2014/05/draw-rounded-corner-line-along-path.html) 

Basically i want as in android java, its already supported for path effects, and there we can change value and make the corner of path smoother and along with that we can basically compose different effects on path.

Just that those skia API's are not available for flutter but surely they are available for android java.

**Comment by imaNNeoFighT** [comment]
Any progress on this issue?

**Comment by porum** [comment]
Any progress on this issue?

**Comment by AlaaAlHallaq** [comment]
Any progress here ! @darshankawar

**Comment by talamaska** [comment]
Hi,

I've read other issues threads and seen some explanations about why the team didn't add certain effects, even though they are already exposed in Skia. The main arguments against adding them was performance.

Now that we are on the path to eventually replacing Skia with the more performant Impeller on all platforms, I'm wondering if these performance arguments are still valid. I think this will be a very valuable addition to the Canvas apis. For example, I'm not sure how this can be achieved in flutter. 
[android gooey effect](https://laewoong.github.io/android-gooey-effect/)
